### PR TITLE
Adds Code and Sprites for Cybernetic Stomachs

### PR DESCRIPTION
**Pull request by Alterist**
## About The Pull Request

Adds 3 tier of Cybernetic Stomachs, makes them researchable by R&D and buildable in the Medical Lathe. Each tier reduces disgust slightly better than the one before it.
https://i.imgur.com/BxSqFzF.png
## Why It's Good For The Game

There are replaceable cybernetic organs for every important organ in the body EXCEPT for stomachs, this PR corrects that.
This allows the Medical Department to be a bit better in treating crewmembers that have been dead for a long period of time or suffered massive trauma and restores their quality of life without relying on nutriment pumps or stomach transplants from other bodies.

## Changelog
:cl:
add: Added 3 tier of Cybernetic Stomachs, researchable by R&D and buildable in the Medical Lathe.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/stomach.dm [hunk 1]
```diff
@@ -128,3 +128,37 @@
...
+
+/obj/item/organ/stomach/cybernetic/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	if(world.time > severe_cooldown) //So we cant just spam emp to kill people.
+		owner.vomit()
+		severe_cooldown = world.time + 10 SECONDS
+	if(prob(emp_vulnerability/severity))	//Chance of permanent effects
+		organ_flags |= ORGAN_SYNTHETIC_EMP //Starts organ faliure - gonna need replacing soon.
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach.dm:164]:
```suggestion
		organ_flags |= ORGAN_SYNTHETIC_EMP //Starts organ faliure - gonna need replacing soon.

```
Linter things

File: code/modules/surgery/organs/stomach.dm [hunk 2]
```diff
@@ -128,3 +128,37 @@
...
+	desc = "An upgraded version of the cybernetic stomach, designed to improve further upon organic stomachs. Handles disgusting food very well."
+	maxHealth = 2 * STANDARD_ORGAN_THRESHOLD
+	disgust_metabolism = 3
+	emp_vulnerability = 20
+
+/obj/item/organ/stomach/cybernetic/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	if(world.time > severe_cooldown) //So we cant just spam emp to kill people.
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach.dm:160]:
```suggestion
	if(!COOLDOWN_FINISHED(src, severe_cooldown)) //So we cant just spam emp to kill people.
```
Just a little macro

File: code/modules/surgery/organs/stomach.dm [hunk 3]
```diff
@@ -128,3 +128,37 @@
...
+		owner.vomit()
+		severe_cooldown = world.time + 10 SECONDS
```
Comment (Rohesie) [inline code/modules/surgery/organs/stomach.dm:162]:
```suggestion
		COOLDOWN_START(src, severe_cooldown, 10 SECONDS)
```

**Comment by Tlaltecuhtli** [comment]
kinda boring effects why not have higher tiers have a chance of spawn mitogen metabolism factor whenever you eat

**Comment by Doosteroni** [comment]
> kinda boring effects why not have higher tiers have a chance of spawn mitogen metabolism factor whenever you eat

Meh, I think we should leave it as is

**Comment by Tlaltecuhtli** [comment]
why would i ever use it then, as having a cyber stomach is worse than not having one (you can get empd)

**Review by Rohesie — DISMISSED** [review]

**Review by Twaticus — APPROVED** [review]

File: code/modules/surgery/organs/stomach.dm [hunk 4]
```diff
@@ -128,3 +128,37 @@
...
+	disgust_metabolism = 3
+	emp_vulnerability = 20
+
+/obj/item/organ/stomach/cybernetic/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	if(!COOLDOWN_FINISHED(src, severe_cooldown)) //So we cant just spam emp to kill people.
+		owner.vomit()
+		COOLDOWN_START(src, severe_cooldown, 10 SECONDS)
```
Comment (kriskog) [inline code/modules/surgery/organs/stomach.dm:162]: Run a stunless vomit for this please. 
vomit(stun = FALSE)

**Review by kriskog — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/stomach.dm (hunk 4 above)
Comment (Alterist) [reply code/modules/surgery/organs/stomach.dm]: Done

**Pull request merged**